ibis/backends/datafusion/tests/test_udf.py

```diff
@@ -0,0 +1,43 @@
+import pandas.testing as tm
+import pyarrow.compute as pc
```

Comment:
does datafusion have a min required pyarrow version? is this indicated somewhere? are you testing on this version? (or master or both)?

Reply:
https://github.com/datafusion-contrib/datafusion-python/blob/main/pyproject.toml#L47

This is almost certainly an untested and invalid claim (that `pyarrow==1` will work with any recent version of datafusion), and it's not ibis's problem to do CI for the datafusion project. This isn't a blocker to merge this PR.

Reply:
sure but we need a min version of pyarrow here (for datafusion), which IS a blocker (well, certainly can merge and do a followup is ok too)

Reply:
I think I'm not communicating effectively here.

There is no knowable minimum version right now. The best can do is say the minimum version is the version of pyarrow as of this PR.

I propose that we continue on as is in the interest of merging this PR unless there's a specific alternative to consider.